File: KCD/HMResetSortieCommand.m
//
//  HMResetSortieCommand.m
//  KCD
//
//  Created by Hori,Masaki on 2014/12/07.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMResetSortieCommand.h"

#import "HMTemporaryDataStore.h"
#import "HMKCBattle.h"


@implementation HMResetSortieCommand

- (void)execute
{
	HMTemporaryDataStore *store = [HMTemporaryDataStore oneTimeEditor];
	NSManagedObjectContext *moc = store.managedObjectContext;
	
	NSError *error = nil;
	NSArray<HMKCBattle *> *battles = [store objectsWithEntityName:@"Battle"
														predicate:nil
															error:NULL];
	if(error) {
		[self log:@"%s error: %@", __PRETTY_FUNCTION__, error];
		return;
	}
	for(HMKCBattle *object in battles) {
		[moc deleteObject:object];
	}
}

@end

File: KCD/HMResetSortieCommand.swift
//
//  HMResetSortieCommand.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/15.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Cocoa

/// 出撃よりの帰還時に出撃内容を消す
class HMResetSortieCommand: HMJSONCommand {
	override func execute() {
		let store = HMTemporaryDataStore.oneTimeEditor()
		var error: NSError? = nil
		let array = store.objectsWithEntityName("Battle", sortDescriptors: nil, predicate: nil, error: &error)
		if error != nil {
			NSLog("\(__FUNCTION__) error: \(error!)")
			return
		}
		let moc = store.managedObjectContext!
		for object in array {
			moc.deleteObject(object as NSManagedObject)
		}
	}
}
